apps/microtvm/zephyr/template_project/microtvm_api_server.py

```diff
@@ -578,49 +623,78 @@ def generate_project(self, model_library_format_path, standalone_crt_dir, projec
     def build(self, options):
         BUILD_DIR.mkdir()
 
-        check_call(["cmake", ".."], cwd=BUILD_DIR)
+        zephyr_board = _find_board_from_cmake_file(API_SERVER_DIR / CMAKELIST_FILENAME)
+        emu_platform = _find_platform_from_cmake_file(API_SERVER_DIR / CMAKELIST_FILENAME)
+
+        env = dict(os.environ)
+        if self._is_fvp(zephyr_board, emu_platform == "armfvp"):
+            env["ARMFVP_BIN_PATH"] = str(API_SERVER_DIR / "fvp-hack")
```

Comment:
The `FVP_Corstone_SSE-300_Ethos-U55` was not an executable on the CI and the tests were failing. I had to explicitly change that with `chmod`. It was OK on my local machine though!

Reply:
also, I checked and the str cast is needed.

Reply:
hmm got it. odd. yeah it's ok on my local machine as well...

Reply:
let me see if I can understand what's happening, this is odd.

Reply:
@driazati Any clue why this file here: https://github.com/apache/tvm/blob/ebd98ce8215eb7499ead0d509457fd432eca7d32/apps/microtvm/zephyr/template_project/fvp-hack/FVP_Corstone_SSE-300_Ethos-U55  (which is indeed an executable) could turn into an non executable  when copied inside the CI container as @mkatanbaf said above? It's copied this way to the new location:

https://github.com/apache/tvm/pull/12125/files#diff-e9070632c64e33062450a5c887a7a6b683d7294064584407aeff7aef24031dc1R561

Reply:
If the script gets sent between a build and a test step in Jenkins it goes through AWS S3 which doesn't have any notion of file permissions, so we add them back in here: https://github.com/apache/tvm/blob/main/ci/jenkins/Build.groovy.j2#L36-L43. Changing that is a little tricky since those changes wont be run in CI unless the source branch is under apache/tvm and not from a fork, so keeping the chmod in code here seems reasonable for now. We can follow up with a change that cleans it up and moves it out to `Build.groovy.j2`

Reply:
@mkatanbaf Could you please add a comment on top of the `chmod` code explaining it's a CI requirement as per David's explanation?

Reply:
done